xbmc/guilib/GUIInfoTypes.h

```diff
@@ -125,6 +126,27 @@ class CGUIInfoLabel
    */
   static std::string ReplaceAddonStrings(const std::string &label);
 
+  typedef boost::function<std::string(const std::string&)> StringReplacerFunc;
+
+  /*!
+   \brief Replaces instances of $strKeyword[value] with the appropriate resolved string
+   \param strInput text to replace
+   \param strKeyword keyword to look for
+   \param func function that does the actual replacement of each bracketed value found
+   \param strOutput the output string
+   \return whether anything has been replaced.
+   */
+  static bool ReplaceDollarKeywordStrings(const std::string &strInput, const std::string &strKeyword, const StringReplacerFunc &func, std::string &strOutput);
```

Comment:
It would be great if you revert the changes to the method signature (leave it private, do not rename it.)
Instead create a convenience method `ReplaceParams(..)` as we have for addon / localize.